das/coordinator.go

```diff
@@ -76,6 +81,12 @@ func (sc *samplingCoordinator) run(ctx context.Context, cp checkpoint) {
 			}
 		case res := <-sc.resultCh:
 			sc.state.handleResult(res)
+
+			// totalSampledFromWorker is the amount of successfully
+			// sampled headers from the worker
+			totalSampledFromWorker := int64(res.To-res.From) - int64(len(res.failed))
+			sc.metrics.recordTotalSampled(ctx, totalSampledFromWorker)
```

Comment:
resultsCh is working with sampling ranges. There is`observeSample` func, that is called every time block is sampled.  You could get much more granular results by updating totalSampled inside `observeSample`.  Would also cleanup coordinator a bit.